crates/shared/src/recent_block_cache.rs

```diff
@@ -397,18 +446,40 @@ mod tests {
 
     #[async_trait::async_trait]
     impl CacheFetching<TestKey, TestValue> for FakeCacheFetcher {
-        async fn fetch_values(&self, _: HashSet<TestKey>, _: Block) -> Result<Vec<TestValue>> {
-            Ok(self.0.lock().unwrap().clone())
+        async fn fetch_values(
+            &self,
+            requested: HashSet<TestKey>,
+            _: Block,
+        ) -> Result<Vec<TestValue>> {
+            let fetched = self
+                .0
+                .lock()
+                .unwrap()
+                .iter()
+                .filter(|value| requested.contains(&TestKey(value.key)))
+                .cloned()
+                .collect();
+            Ok(fetched)
+        }
+    }
+
+    impl FakeCacheFetcher {
+        pub fn new(values: Vec<TestValue>) -> Self {
+            Self(Arc::new(Mutex::new(values)))
         }
     }
 
     fn test_keys(keys: impl IntoIterator<Item = usize>) -> impl Iterator<Item = TestKey> {
         keys.into_iter().map(TestKey)
     }
 
-    #[test]
-    fn marks_recently_used() {
-        let fetcher = FakeCacheFetcher::default();
+    #[tokio::test]
```

Comment:
These tests had to be marked as async tests because `RequestSharing` that gets now used under the hood spawns a task which would panic on a sync test.